openshift_performance/ose3_perf/scripts/build_test_1.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/env python
+import re
+import subprocess
+import json
+import time
+import datetime
+import random
+import sys
+from optparse import OptionParser
+# pip install futures
+from concurrent.futures import ThreadPoolExecutor, wait
+# https://github.com/wroberts/pytimeparse
+# pip install pytimeparse
+from pytimeparse.timeparse import timeparse
+import logging
+
+
+def run(cmd, config=""):
+
+    if config:
+        cmd = "KUBECONFIG=" + config + " " + cmd
+    result = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
+    return result
+
+
+def login(url, user, passwd):
+    run("oc login --insecure-skip-tls-verify=true -u " + user + " -p " + passwd
+        + " " + url)
+    # run("oc login https://api.dev-preview-int.openshift.com "
+    #    "--token=yPeDkR0b0ESwbX2WlC4AGWqQIHJHIeRbrEOOua9HEoQ")
+
+
+def run_build(build_def):
+    build_regex = re.compile("build \"(.*)\" started")
+
+    namespace = build_def["namespace"]
+    name = build_def["name"]
+    build_name = "UNKNOWN"
+
+    try:
+
+        build_name = run("oc start-build -n " + namespace + " " + name).rstrip()
+        # 3.2 Comment out following line
+        build_name = build_regex.search(build_name).group(1)
+        global_build_status[namespace + ":" + build_name] = STATUS_STARTED
+        logger.info("Build is: " + namespace + ":" + build_name)
+
+    except subprocess.CalledProcessError as err:
+        if "UNKNOWN" != build_name:
+            global_build_status[namespace + ":" + build_name] = STATUS_ERROR
+        logger.error("Command failed:  tproject=" + namespace + ",cmd=" +
+                     err.cmd + ", retcode=" + str(err.returncode) + ", output="
+                     + err.output)
+    except Exception as e:
+        logger.error("cannot get build_name from: " + build_name)
+        logger.error(e)
+
+
+def check_build_status(executor):
+    logger.info("check_build_status ...")
+    # wait the first build is started
+    time.sleep(20)
+    while not all_builds_completed():
+        try:
+            time.sleep(10)
+            result = run("oc get build --all-namespaces --no-headers")
```

Comment:
My previous comment in our conversation about this being a concern was invalid.   oc get build... runs sync in the main thread, so even if it takes > 10 seconds due to large numbers of completed builds it will not cause issues.

Reply:
Understood.